### PR TITLE
fs: Added fsCreateSaveDataFileSystem(), fsDeleteSaveDataFileSystem() and fsDeleteSaveDataFileSystemBySaveDataAttribute().

### PR DESCRIPTION
**Pull request by DarkMatterCore**
These are needed to easily generate custom `FsSaveDataType_Cache` savedata entries.

Unlike normal savedata, cache savedata size can't be retrieved from the NACP, which means it's necessary to calculate its required size. One could then create a cache savedata entry larger than the calculated size, which is simpler than having to resize the existent save each time new data is written to it.

### CODE REVIEW

**Comment by SciresM** [comment]
PR has mixed tabs/spaces which should be fixed before merge.

What's the use case for cache savedata for homebrew? Not necessarily objecting to merge right now (since we have other savedata creation/management stuff), but also not sure that I actually understand what this is for.

**Comment by DarkMatterCore** [comment]
@SciresM Done, thanks for spotting that.

There are games like the old Doom re-releases that use cache savedata entries to store WAD packages. It's possible to replace these WAD packages to play custom levels.

File: nx/source/services/fs.c
Comment (yellows8) [inline nx/source/services/fs.c:232]: Pass application_id directly, instead of in.

File: nx/include/switch/services/fs.h
Comment (SciresM) [inline nx/include/switch/services/fs.h:96]: This is "nn::fs::SaveDataMetaInfo", rename to FsSaveDataMetaInfo.

Rename "unk" to "reserved", since that's what the field is.

**Review by SciresM — CHANGES_REQUESTED** [review]
FsSaveMeta should use the real typename.

File: nx/include/switch/services/fs.h
Comment (SciresM) [reply nx/include/switch/services/fs.h:96]: Also, consider renaming "meta_index" to "meta_type", since this is nn::fs::SaveDataMetaType (enum).

Enum values are:

```
None             = 0,
Thumbnail        = 1,
ExtensionContext = 2,
```

Comment (SciresM) [reply nx/include/switch/services/fs.h:96]: Honestly, rename the fields as follows:

```
meta_file_size -> size
meta_index -> type
unk -> reserved
```

**Comment by DarkMatterCore** [comment]
Went ahead and added `fsDeleteSaveDataFileSystemBySaveDataAttribute()` as well. Figured it may be just as useful as `fsDeleteSaveDataFileSystem()`.

**Review by fincs — APPROVED** [review]

**Review by yellows8 — APPROVED** [review]

**Pull request merged**
